include/dfrn.php

```diff
@@ -1045,4 +1067,1311 @@ function deliver($owner,$contact,$atom, $dissolve = false) {
+
+		// update last-child if it changes
+		if($item["last-child"] AND ($item["last-child"] != $current["last-child"])) {
+			$r = q("UPDATE `item` SET `last-child` = 0, `changed` = '%s' WHERE `parent-uri` = '%s' AND `uid` = %d",
+				dbesc(datetime_convert()),
+				dbesc($item["parent-uri"]),
+				intval($importer["importer_uid"])
+			);
+			$r = q("UPDATE `item` SET `last-child` = %d , `changed` = '%s' WHERE `uri` = '%s' AND `uid` = %d",
+				intval($item["last-child"]),
+				dbesc(datetime_convert()),
+				dbesc($item["uri"]),
+				intval($importer["importer_uid"])
+			);
+		}
+		return $changed;
+	}
+
+	/**
+	 * @brief Detects the entry type of the item
+	 *
+	 * @param array $importer Record of the importer user mixed with contact of the content
+	 * @param array $item the new item record
+	 *
+	 * @return int Is it a toplevel entry, a comment or a relayed comment?
+	 */
+	private function get_entry_type($importer, $item) {
+		if ($item["parent-uri"] != $item["uri"]) {
+			$community = false;
+
+			if($importer["page-flags"] == PAGE_COMMUNITY || $importer["page-flags"] == PAGE_PRVGROUP) {
+				$sql_extra = "";
+				$community = true;
+				logger("possible community action");
+			} else
+				$sql_extra = " AND `contact`.`self` AND `item`.`wall` ";
+
+			// was the top-level post for this action written by somebody on this site?
+			// Specifically, the recipient?
+
+			$is_a_remote_action = false;
+
+			$r = q("SELECT `item`.`parent-uri` FROM `item`
+				WHERE `item`.`uri` = '%s'
+				LIMIT 1",
+				dbesc($item["parent-uri"])
+			);
+			if($r && count($r)) {
+				$r = q("SELECT `item`.`forum_mode`, `item`.`wall` FROM `item`
+					INNER JOIN `contact` ON `contact`.`id` = `item`.`contact-id`
+					WHERE `item`.`uri` = '%s' AND (`item`.`parent-uri` = '%s' OR `item`.`thr-parent` = '%s')
+					AND `item`.`uid` = %d
+					$sql_extra
+					LIMIT 1",
+					dbesc($r[0]["parent-uri"]),
+					dbesc($r[0]["parent-uri"]),
+					dbesc($r[0]["parent-uri"]),
+					intval($importer["importer_uid"])
+				);
+				if($r && count($r))
+					$is_a_remote_action = true;
+			}
+
+			// Does this have the characteristics of a community or private group action?
+			// If it's an action to a wall post on a community/prvgroup page it's a
+			// valid community action. Also forum_mode makes it valid for sure.
+			// If neither, it's not.
+
+			if($is_a_remote_action && $community) {
+				if((!$r[0]["forum_mode"]) && (!$r[0]["wall"])) {
+					$is_a_remote_action = false;
+					logger("not a community action");
+				}
+			}
+
+			if ($is_a_remote_action)
+				return DFRN_REPLY_RC;
+			else
+				return DFRN_REPLY;
+
+		} else
+			return DFRN_TOP_LEVEL;
+
+	}
+
+	/**
+	 * @brief Send a "poke"
+	 *
+	 * @param array $item the new item record
+	 * @param array $importer Record of the importer user mixed with contact of the content
+	 * @param int $posted_id The record number of item record that was just posted
+	 */
+	private function do_poke($item, $importer, $posted_id) {
```

Comment:
Without knowing too much about the internals, is `do_poke` the correct name because all it is doing is sending a notification.